### ISSUE TITLE
xonshrc caching causing errors on startup

### ISSUE THREAD

**Issue by gforsyth** [scopatz/xonsh#761]
`xonsh` is broken for me as of 65467e9

```console
(xonshtest)gil@theo~/git/xonsh HEAD$ xonsh
using cached marshaled rcfile
Traceback (most recent call last):
  File "/home/gil/anaconda/envs/xonshtest/bin/xonsh", line 9, in <module>
    load_entry_point('xonsh==0.2.6', 'console_scripts', 'xonsh')()
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/main.py", line 199, in main
    args = premain(argv)
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/main.py", line 186, in premain
    shell = builtins.__xonsh_shell__ = Shell(**shell_kwargs)
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/shell.py", line 78, in __init__
    self._init_environ(ctx, config, rc)
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/shell.py", line 125, in _init_environ
    self.ctx = xonshrc_context(rcfiles=rc, execer=self.execer)
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/environ.py", line 1192, in xonshrc_context
    exec(ccode, env, None)
TypeError: exec() arg 1 must be a string, bytes or code object
```

Running with `--no-rc` removes the error

```console
(xonshtest)gil@theo~/git/xonsh HEAD$ xonsh --no-rc
Could not find xonsh configuration or run control files.
Would you like to run the xonsh configuration wizard now, yes or no [default: no]? no
(xonshtest)gil@theo~/git/xonsh HEAD$ 
```

**Comment by scopatz** [comment]
Does it work with #757?

**Comment by gforsyth** [comment]
Sadly, no.

```console
(xonshtest)gil@theo~/git/xonsh pr/757$ xonsh
Traceback (most recent call last):
  File "/home/gil/anaconda/envs/xonshtest/bin/xonsh", line 9, in <module>
    load_entry_point('xonsh==0.2.7', 'console_scripts', 'xonsh')()
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/main.py", line 218, in main
    args = premain(argv)
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/main.py", line 204, in premain
    shell = builtins.__xonsh_shell__ = Shell(**shell_kwargs)
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/shell.py", line 80, in __init__
    kwargs.get('cacheall', False))
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/shell.py", line 129, in _init_environ
    self.ctx = xonshrc_context(rcfiles=rc, execer=self.execer)
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/environ.py", line 1162, in xonshrc_context
    run_script_with_cache(rcfile, execer, env)
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/codecache.py", line 135, in run_script_with_cache
    run_compiled_code(ccode, glb, loc, mode)
  File "/home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/codecache.py", line 55, in run_compiled_code
    func(code, glb, loc)
TypeError: exec() arg 1 must be a string, bytes or code object
(xonshtest)gil@theo~/git/xonsh pr/757$ xonsh --no-rc
Could not find xonsh configuration or run control files.
Would you like to run the xonsh configuration wizard now, yes or no [default: no]? no
(xonshtest)gil@theo~/git/xonsh pr/757$ 
```

**Comment by scopatz** [comment]
Would you mind jumping into pdb or something and try to figure out what the type of `code` is?

**Comment by gforsyth** [comment]
Will do.

**Comment by scopatz** [comment]
Thanks!

**Comment by gforsyth** [comment]
```console
> /home/gil/anaconda/envs/xonshtest/lib/python3.5/site-packages/xonsh/codecache.py(56)run_compiled_code()
-> func(code, glb, loc)
(Pdb) type(code)
<class 'NoneType'>
```

**Comment by scopatz** [comment]
Do you have a xonshrc file?

**Comment by scopatz** [comment]
Or is it empty?

**Comment by gforsyth** [comment]
Aha!  I do, and it's not, but I _did_ have an `/etc/xonshrc` file that was empty.  Removing that fixed it

**Comment by scopatz** [comment]
Ahh, ok.  This code should still be fixed up, though :)

**Comment by scopatz** [comment]
Just a simple `if code is None: return` would be good enough.  Would you mind putting in a PR, please, @gforsyth?  I am pretty busy today sadly :(

**Comment by gforsyth** [comment]
No problem -- incoming!

**Comment by gforsyth** [comment]
Resolved in #762
